pkg/cloudwatch/search.go

```diff
@@ -0,0 +1,137 @@
+package cloudwatch
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs/types"
+	"github.com/flanksource/apm-hub/api/logs"
+)
+
+func NewCloudWatchSearchBackend(config *logs.CloudWatchBackendConfig, client *cloudwatchlogs.Client) *cloudWatchSearch {
+	return &cloudWatchSearch{
+		client: client,
+		config: config,
+	}
+}
+
+type cloudWatchSearch struct {
+	client *cloudwatchlogs.Client
+	config *logs.CloudWatchBackendConfig
+}
+
+func (t *cloudWatchSearch) MatchRoute(q *logs.SearchParams) (match bool, isAdditive bool) {
+	return t.config.CommonBackend.Routes.MatchRoute(q)
+}
+
+func (t *cloudWatchSearch) Search(q *logs.SearchParams) (logs.SearchResults, error) {
+	logFilter := &cloudwatchlogs.StartQueryInput{
+		LogGroupName: &t.config.LogGroup,
+		Limit:        ptr(int32(q.Limit)),
+		QueryString:  &t.config.Query,
+	}
+
+	if q.GetStart() != nil {
+		logFilter.StartTime = ptr(q.GetStart().UnixMilli())
+	}
+
+	if q.GetEnd() != nil {
+		logFilter.EndTime = ptr(q.GetEnd().UnixMilli())
+	} else {
+		logFilter.EndTime = ptr(time.Now().UnixMilli()) // end time is a required field
+	}
+
+	var result logs.SearchResults
+	queryOutput, err := t.client.StartQuery(context.Background(), logFilter)
+	if err != nil {
+		return result, err
+	}
+
+	queryResult, err := t.getQueryResults(queryOutput.QueryId)
+	if err != nil {
+		return result, err
+	}
+
+	result.Total = int(queryResult.Statistics.RecordsMatched)
+
+	result.Results = make([]logs.Result, 0, len(queryResult.Results))
+	for _, fields := range queryResult.Results {
+		var event = logs.Result{
+			Labels: t.config.Labels,
+		}
+
+		for _, field := range fields {
+			switch *field.Field {
+			case "@message":
+				event.Message = deref(field.Value)
+			case "@timestamp":
+				event.Time = toRFC339(deref(field.Value))
```

Comment:
@moshloop Do we want to support custom message and timestamp field like we did with elastic search?
https://github.com/flanksource/apm-hub/blob/57cb7c036fab14b8f1db7bf58bcb06edfba93791/api/logs/logs.go#L143